nltk/tokenize/casual.py

```diff
@@ -156,7 +156,7 @@
 
     # Remaining word types:
     r"""
-    (?:[a-z][a-z'\-_]+[a-z])       # Words with apostrophes or dashes.
+    (?:\w[\w'\-_]+\w)              # Words with apostrophes or dashes.
```

Comment:
We ought to be making our assumptions about `casual.py` explicit for regression testing in a file, i.e. in `test/tokenize.doctest`. @fievelk do you have time to do a first pass on this?